docs/source/examples/Detecting Events.mystnb

```diff
@@ -0,0 +1,386 @@
+from poliastro.plotting import StaticOrbitPlotter
+
+from poliastro.core.events import eclipse_function
+from poliastro.core.perturbations import atmospheric_drag_exponential
+from poliastro.core.propagation import func_twobody
+from poliastro.core.spheroid_location import cartesian_to_ellipsoidal
+
+from poliastro.twobody.events import(
+    AltitudeCrossEvent,
+    LatitudeCrossEvent,
+    NodeCrossEvent,
+    PenumbraEvent,
+    UmbraEvent,
+)
+from poliastro.twobody.propagation import cowell
+from poliastro.twobody.orbit import Orbit
+
+from poliastro.util import time_range
+```
+
+## Altitude Crossing Event
+Let's define some natural perturbation conditions for our orbit so that its altitude decreases with time.
+
+```{code-cell} ipython3
+R = Earth.R.to(u.km).value
+
+# Parameters of the body
+C_D = 2.2  # Dimensionless (any value would do)
+A_over_m = ((np.pi / 4.0) * (u.m ** 2) / (100 * u.kg)).to_value(
+    u.km ** 2 / u.kg
+)  # km^2/kg
+
+# Parameters of the atmosphere
+rho0 = rho0_earth.to(u.kg / u.km ** 3).value  # kg/km^3
+H0 = H0_earth.to(u.km).value  # km
+
+def f(t0, u_, k):
+    du_kep = func_twobody(t0, u_, k)
+    ax, ay, az = atmospheric_drag_exponential(
+        t0, u_, k, R=R, C_D=C_D, A_over_m=A_over_m, H0=H0, rho0=rho0
+    )
+    du_ad = np.array([0, 0, 0, ax, ay, az])
+    return du_kep + du_ad
+```
+
+We shall use the `cowell` propagator with the above perturbating conditions and pass the events we want to keep track of, in this case only the `AltitudeCrossEvent`.
+
+```{code-cell} ipython3
+tofs = np.arange(0, 2400, 100) << u.s
+orbit = Orbit.circular(Earth, 150 * u.km)
+
+# Define a threshold altitude for crossing.
+thresh_alt = 50  # in km
+altitude_cross_event = AltitudeCrossEvent(thresh_alt, R)  # Set up the event.
+events = [altitude_cross_event]
+
+rr, _ = cowell(
+    Earth.k,
+    orbit.r,
+    orbit.v,
+    tofs,
+    events=events,
+    f=f,
+)
+
+print(f"The threshold altitude was crossed {altitude_cross_event.last_t} after the orbit's epoch.")
+```
+
+Let's see how did the orbit's altitude vary with time:
+
+```{code-cell} ipython3
+altitudes = np.apply_along_axis(norm, 1, (rr << u.km).value) - Earth.R.to_value(u.km)
+plt.plot(tofs[:len(rr)].to_value(u.s), altitudes);
+plt.title("Altitude variation")
+plt.ylabel("Altitude (in km)");
+plt.xlabel("Time (in s)");
+```
+Refer to the API documentation of the events to check the default values for `terminal` and `direction` and change it as required.
+
+
+## Latitude Crossing Event
+
+Similar to the `AltitudeCrossEvent`, just pass the threshold latitude while instantiating the event.
+
+```{code-cell} ipython3
+orbit = Orbit.from_classical(
+    Earth, 6900 << u.km, 0.63 * u.one, 30 * u.deg, 0 * u.deg, 0 * u.deg, 0 * u.deg
+)
+```
+
+```{code-cell} ipython3
+thresh_lat = 20 * u.deg
+latitude_cross_event = LatitudeCrossEvent(orbit, thresh_lat, terminal=True)
+events = [latitude_cross_event]
+
+tofs = np.arange(0, 2000, 10) << u.s
```

Comment:
An the time of flight is given as a multiple of the orbital period to ensure enough propagation time:

```python
tofs = np.arange(0, 20 * orbit.period.to(u.s).value, 150) << u.s
```